nixos/modules/services/networking/sickbeard.nix

```diff
@@ -0,0 +1,79 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.sickbeard;
+  sickbeard = cfg.package;
+in
+{
+
+  ###### interface
+
+  options = {
+    services.sickbeard = {
+      enable = mkOption {
+        default = false;
+        description = "Whether to enable the sickbeard server.";
+      };
+      package = mkOption {
+        type = types.package;
+        default = pkgs.sickbeard;
+        example = literalExample "pkgs.sickrage";
+        description =''
+          Enable <literal>pkgs.sickrage</literal> or <literal>pkgs.sickgear</literal>
+          as an alternative to SickBeard
+        '';
+      };
+      dataDir = mkOption {
+        default = "/var/lib/sickbeard";
+        description = "Path where to store data files.";
+      };
+      configFile = mkOption {
```

Comment:
Would love to see a type = types.str; here as well.

Reply:
types.str or types.path?

Reply:
Based on a quick search I believe types.path can only refer to an actual directory and not a file, but I'm not certain of this.

I believe the answer is types.str but would appreciate being corrected by anyone of this is not the case.

Reply:
I think (according to [this](https://github.com/NixOS/nixpkgs/blob/80d7a4c87281b4fb41105e4bd22d4d600fdbf153/lib/types.nix#L236)) it means it's an absolute path? Which I think would be better here as, at least with sickgear, it seems like it puts relative paths relative to /. So it was failing when I tried it with types.str with a relative path. At least types.path enforces an absolute path

Reply:
Good to know, thanks!